silkrpc/ethbackend/remote_backend.cpp

```diff
@@ -91,6 +93,33 @@ boost::asio::awaitable<uint64_t> RemoteBackEnd::net_peer_count() {
     co_return count;
 }
 
+boost::asio::awaitable<std::vector<NodeInfo>> RemoteBackEnd::engine_node_info() {
+    std::vector<NodeInfo> node_info_list;
+    const auto start_time = clock_time::now();
+    UnaryRpc<&::remote::ETHBACKEND::StubInterface::AsyncNodeInfo> ni_rpc{*stub_, grpc_context_};
+    const auto reply = co_await ni_rpc.finish_on(executor_, ::remote::NodesInfoRequest{});
+    for (int i = 0; i < reply.nodesinfo_size(); i++) {
+        NodeInfo node_info;
+        const auto backend_node_info = reply.nodesinfo(i);
+        node_info.id = backend_node_info.id();
+        node_info.name = backend_node_info.name();
+        node_info.enode = backend_node_info.enode();
+        node_info.enr = backend_node_info.enr();
+        node_info.listenerAddr = backend_node_info.listeneraddr();
+        const auto protocols = backend_node_info.protocols();
+        auto protocols_json = nlohmann::json::parse(protocols, nullptr, /* allow_exceptions = */ false);
+        node_info.protocols = decode_protocols(protocols_json);
```

Comment:
I suggest not to decode the protocol JSON fragment here if the same exact JSON object will be present in the `erigon_nodeInfo` JSON: you can just keep it as plain string and move the `nlohmann::json::parse` call to the place where JSON serialization of `erigon_nodeInfo` result happens.

Reply:
ok